lib/logstash/codecs/cef.rb

```diff
@@ -75,6 +75,9 @@ class LogStash::Codecs::CEF < LogStash::Codecs::Base
" => "app",  "deviceCustomIPv6Address1" => "c6a1", "deviceCustomIPv6Address1Label" =>  "c6a1Label",  "deviceCustomIPv6Address2" => "c6a2",  "deviceCustomIPv6Address2Label" => "c6a2Label",  "deviceCustomIPv6Address3" =>  "c6a3",  "deviceCustomIPv6Address3Label" => "c6a3Label",  "deviceCustomIPv6Address4" => "c6a4",  "deviceCustomIPv6Address4Label" =>  "c6a4Label",  "deviceEventCategory" => "cat",  "deviceCustomFloatingPoint1" => "cfp1",  "deviceCustomFloatingPoint1Label" => "cfp1Label",  "deviceCustomFloatingPoint2" => "cfp2",  "deviceCustomFloatingPoint2Label" =>  "cfp2Label",  "deviceCustomFloatingPoint3" => "cfp3",  "deviceCustomFloatingPoint3Label" => "cfp3Label",  "deviceCustomFloatingPoint4" => "cfp4",  "deviceCustomFloatingPoint4Label" =>  "cfp4Label",  "deviceCustomNumber1" => "cn1",  "deviceCustomNumber1Label" =>  "cn1Label",  "deviceCustomNumber2" => "cn2",  "deviceCustomNumber2Label" =>  "cn2Label", "deviceCustomNumber3" => "cn3",  "deviceCustomNumber3Label" =>  "cn3Label",  "baseEventCount" => "cnt",  "deviceCustomString1" => "cs1",  "deviceCustomString1Label" => "cs1Label",  "deviceCustomString2" => "cs2",  "deviceCustomString2Label" => "cs2Label",  "deviceCustomString3" => "cs3",  "deviceCustomString3Label" => "cs3Label",  "deviceCustomString4" => "cs4",  "deviceCustomString4Label" => "cs4Label",  "deviceCustomString5" => "cs5",  "deviceCustomString5Label" => "cs5Label", "deviceCustomString6" => "cs6",  "deviceCustomString6Label" => "cs6Label",  "destinationHostName" => "dhost",  "destinationMacAddress" => "dmac",  "destinationNtDomain" => "dntdom",  "destinationProcessId" => "dpid",  "destinationUserPrivileges" => "dpriv",  "destinationProcessName" => "dproc",  "destinationPort" => "dpt",  "destinationAddress" => "dst",  "destinationUserId" => "duid",  "destinationUserName" => "duser",  "deviceAddress" => "dvc",  "deviceHostName" =>  "dvchost", "deviceMacAddress" => "dvcmac", "deviceProcessId" => "dvcpid",  "endTime" => "end",  "fileName" => "fname",  "fileSize" => "fsize",  "bytesIn" =>  "in",  "message" => "msg",  "bytesOut" => "out",  "eventOutcome" => "outcome",  "transportProtocol" => "proto",  "requestUrl" => "request",  "deviceReceiptTime" => "rt",  "sourceHostName" => "shost",  "sourceMacAddress" =>  "smac",  "sourceNtDomain" => "sntdom",  "sourceProcessId" => "spid",  "sourceUserPrivileges" => "spriv",  "sourceProcessName"=> "sproc",  "sourcePort" => "spt",  "sourceAddress" => "src",  "startTime" => "start",  "sourceUserId" => "suid",  "sourceUserName" => "suser",  "agentHost" =>  "ahost",  "agentReceiptTime" => "art",  "agentType" => "at",  "agentId" =>  "aid",  "cefVersion" => "_cefVer",  "agentAddress" => "agt",  "agentVersion" =>  "av",  "agentTimeZone" => "atz",  "destinationTimeZone" => "dtz",  "sourceLongitude" => "slong",  "sourceLatitude" => "slat",  "destinationLongitude" => "dlong",  "destinationLatitude" => "dlat",  "categoryDeviceType" => "catdt",  "managerReceiptTime" => "mrt",  "agentMacAddress" => "amac" }  
```

Comment:
if `REVERSE_MAPPINGS` is the exact inverse of `MAPPINGS` and `MAPPINGS` values are unique then you can use 
```rb
REVERSE_MAPPINGS = MAPPINGS.invert
```
instead of manually inverting it.

Reply:
I just reviewed the fields and realized that there is 1 missing field in the original MAPPINGS:
"dvcmac" => deviceMacAddress

That is the only difference.
I believe it should be there.
I will make the changes and simply invert the MAPPINGS for REVERSE_MAPPINGS.